controllers/openstackcluster_controller.go

```diff
@@ -912,9 +912,11 @@ func (r *OpenStackClusterReconciler) SetupWithManager(ctx context.Context, mgr c
 }
 
 func handleUpdateOSCError(openstackCluster *infrav1.OpenStackCluster, message error) {
-	err := capierrors.UpdateClusterError
-	openstackCluster.Status.FailureReason = &err
-	openstackCluster.Status.FailureMessage = ptr.To(message.Error())
+	if !openstackCluster.Status.Ready {
```

Comment:
I'd prefer to not call this method for a transient error in the first place. This would prevent anything from being fatal after the cluster is ready, which probably isn't right either.

Reply:
Ok moved the ready check outside of the function. 

But there a 4 place i am not sure what to do:

fist case:  
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/controllers/openstackcluster_controller.go#L815-L822
Here some one added a terminal error to the function, but inside the function it never returns with terminalError true

the second case: 
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/controllers/openstackcluster_controller.go#L815-L822
In this function someone selected exactly one error already, but did not check against ready. 

In both cases i am not sure if we need to add the ready state check on top?